### PR TITLE
Elasticsearch

### PR DESCRIPTION
**Pull request by kfrancoi**
This PR adds two things : 
- Dockerization of the project
- Add Elasticsearch backend

### CODE REVIEW

File: test/test_elasticsearch.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env python
+
+'''Make sure the Mongo client is sane'''
```
Comment (aborsu) [inline test/test_elasticsearch.py:3]: Mongo client?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pymongo == 2.3
```
Comment (aborsu) [inline requirements.txt:1]: Are you using this?

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+python-dateutil
+gevent
```
Comment (aborsu) [inline requirements.txt:3]: idem

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+elasticsearch ~= 2.1.0
+-e git://github.com/Sagacify/simhash-py.git#egg=simhash
```
Comment (aborsu) [inline requirements.txt:5]: idem

File: simhash_db/elasticsearch_client.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+import struct
+import time
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import ConnectionError
+from . import BaseClient
+import calendar
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import dateutil.parser
```
Comment (aborsu) [inline simhash_db/elasticsearch_client.py:13]: You are not using the following imports
from elasticsearch.exceptions import ConnectionError
import calendar
from datetime import datetime
from dateutil.relativedelta import relativedelta
import dateutil.parser

Comment (kfrancoi) [inline simhash_db/elasticsearch_client.py:13]: Bien vu!

**Pull request merged**